pandapower/control/util/diagnostic.py

```diff
@@ -64,54 +64,102 @@ def control_diagnostic(net, respect_in_service=True):
+            warnings.warn(f"There are some transformers in the {trafo_table} table with not all "
+                          f"characteristics populated in the trafo_characteristic_table.", category=UserWarning)
+        # check tap_dependency_table & id_characteristic_table column types
+        if net[trafo_table]['tap_dependency_table'].dtype != 'bool':
+            warnings.warn(f"The tap_dependency_table column in the {trafo_table} table is not of bool type.",
+                          category=UserWarning)
+        if net[trafo_table]['id_characteristic_table'].dtype != 'Int64':
+            warnings.warn(f"The id_characteristic_table column in the {trafo_table} table is not of Int64 type.",
+                          category=UserWarning)
+        # check if all id_characteristic_table values are present in id_characteristic column
+        # of trafo_characteristic_table
+        if not net[trafo_table]['id_characteristic_table'].dropna().isin(
+                net["trafo_characteristic_table"]['id_characteristic']).all():
+            warnings.warn(f"Not all id_characteristic_table values in the {trafo_table} table are present "
+                          f"in id_characteristic column of trafo_characteristic_table.", category=UserWarning)
+
 
-            # chack if any characteristics have value at the neutral point that deviates from the transformer parameter
-            variable = col.replace("_characteristic", "")
-            for element_index in net[trafo_table].index[tap_dependent_impedance]:
-                tap_neutral = net[trafo_table].tap_neutral.fillna(0).at[element_index]
-                s_id = net[trafo_table][col].at[element_index]
-                if pd.isnull(s_id):
-                    continue
-                s = net.characteristic.object.at[s_id]
-                s_val = s(tap_neutral)
-                var_val = net[trafo_table].at[element_index, variable]
-                if not np.isclose(s_val, var_val, rtol=0, atol=1e-6):
-                    logger.warning(
-                        f"The characteristic value of {s_val} at the neutral tap position "
-                        f"{tap_neutral} does not match the value {var_val} of {variable} for the "
-                        f"{trafo_table} with index {element_index} (deviation of {s_val-var_val})")
+def shunt_characteristic_table_diagnostic(net):
+    logger.info("Checking shunt characteristic table")
+    if "shunt_characteristic_table" not in net:
+        logger.info("No shunt characteristic table found")
+        return False
+    cols = ["id_characteristic", "step", "q_mvar", "p_mw"]
+    if len(net["shunt"]) == 0 or \
+            not all(col in net["shunt"] for col in ['id_characteristic_table', 'step_dependency_table']) or \
+            (not net["shunt"]['id_characteristic_table'].notna().any() and
+             not net["shunt"]['step_dependency_table'].any()):
+        logger.info("No %s with step-dependent characteristics found." % "shunt")
```

Comment:
you can directly write shunt in the string. like **logger.info("No shunt with step-dependent characteristics found."**